### PR TITLE
ISSUE-2036: Using node_name for container notifications

### PR DESCRIPTION
**Pull request by varunsharma0286**
Fixes #
https://github.com/deepfence/enterprise-roadmap/issues/2036

Changes proposed in this pull request:
- Using the field `node_name` to the query for checking notifications for the container.

### CODE REVIEW

File: deepfence_server/reporters/scan/scan_reporters.go [hunk 1]
```diff
@@ -500,19 +500,30 @@ func GetScansList(ctx context.Context, scanType utils.Neo4jScanType, nodeIDs []m
...
+				nodeNamesStr = append(nodeNamesStr, nodeID.NodeName)
+			}
 		}
+
 		query = `
 			MATCH (m:` + string(scanType) + `) -[:SCANNED]-> (n)
-			WHERE n.node_id IN $node_ids
+			WHERE (n.node_id IN $node_ids`
+		if len(nodeNamesStr) > 0 {
+			query = query + ` OR n.node_name IN $node_names`
```
Comment (noboruma) [inline deepfence_server/reporters/scan/scan_reporters.go:520]: As linted ;)

Comment (varunsharma0286) [reply deepfence_server/reporters/scan/scan_reporters.go]: Done 👍

File: deepfence_server/model/scans.go [hunk 2]
```diff
@@ -65,6 +65,7 @@ type ScanTriggerCommon struct {
 type NodeIdentifier struct {
 	NodeID   string `json:"node_id" required:"true"`
 	NodeType string `json:"node_type" required:"true" enum:"image,host,container,cloud_account,cluster,registry,pod"`
+	NodeName string `json:"node_name" required:"false"`
 }
```
Comment (noboruma) [inline deepfence_server/model/scans.go:70]: Now I am not sure we should do that..
Can't we run two queries:
- One with the NodeName as condition to find all the NodeIDs
- The next query will do the actual search using those nodeIDs

File: deepfence_worker/cronjobs/notification.go [hunk 3]
```diff
@@ -389,7 +464,8 @@ func processIntegration[T any](ctx context.Context, task *asynq.Task, integratio
 	}
 	log.Info().Msgf("%s Total Time taken for integration %s: %d", integrationRow.Resource,
 		integrationRow.IntegrationType, time.Since(startTime).Milliseconds())
-	log.Debug().Msgf("Time taken for neo4j_query_1: %d", neo4j_query_1)
+	log.Debug().Msgf("Time taken for neo4j_query_nc: %d", neo4j_query_nc)
+	log.Debug().Msgf("Time taken for neo4j_query_c: %d", neo4j_query_c)
 	log.Debug().Msgf("Time taken for neo4j_query_2: %d", totalQueryTime)
 	log.Debug().Msgf("Time taken for sending data : %d", totalSendTime)
```
Comment (noboruma) [inline deepfence_worker/cronjobs/notification.go:470]: We don't need any of these, we capture time for every neo4j queries in our system.
The traces are available via jaegger.

File: deepfence_worker/cronjobs/notification.go [hunk 4]
```diff
@@ -307,22 +307,97 @@ func processIntegration[T any](ctx context.Context, task *asynq.Task, integratio
...
+	}
+
+	var nodeIDs, nodeTypes []interface{}
+	nodeTypeSet := make(map[string]bool)
+	for _, v := range filters.NodeIds {
+		if v.NodeType == "container" {
+			continue
+		}
+		nodeIDs = append(nodeIDs, v.NodeID)
+		if _, ok := nodeTypeSet[v.NodeType]; !ok {
```
Comment (noboruma) [inline deepfence_worker/cronjobs/notification.go:336]: Can we encapsulate the whole inside a nice `integrationFilters2SearchScanReqs` function?

**Review by noboruma — APPROVED** [review]

**Pull request merged**